lib/core/src/plugin.rs

```
@@ -0,0 +1,56 @@
+use anyhow::{bail, Result};
+use maybe_sync::{MaybeSend, MaybeSync};
+
+use crate::{error::SdkResult, persist::Persister, sdk::LiquidSdk};
+use sdk_common::utils::Arc;
+
+pub struct PluginStorage {
+    plugin_id: String,
+    persister: std::sync::Arc<Persister>,
+}
+
+impl PluginStorage {
+    pub(crate) fn new(persister: std::sync::Arc<Persister>, plugin_id: String) -> Result<Self> {
+        if plugin_id.is_empty() {
+            log::error!("Plugin ID cannot be an empty string!");
+            bail!("Plugin ID cannot be an empty string!");
+        }
+
+        Ok(Self {
+            persister,
+            plugin_id,
+        })
+    }
+
+    pub(crate) fn scoped_key(&self, key: String) -> String {
+        format!("{}-{}", self.plugin_id, key)
+    }
+
+    pub fn set_item(&self, key: String, value: String) -> SdkResult<()> {
```

Would it make sense to have a PluginStorage-specific result here instead of `SdkResult`? Same for the other public methods

---

Tbh the only error variant that can occur is if there's a problem connecting and writing to the database. Not sure whether it's worth creating a new error type for, but I'm not strongly against it either. Maybe we could also just return a String error "could not write to database".